docs/adr/00xx-storage-allow-object_ids.md

```diff
@@ -0,0 +1,52 @@
+---
+status: "proposed"
+---
+# Add support for specifying object_ids when calling the storage endpoint
+
+## Context and Problem Statement
+
+Currently the TAMS API generates the `object_id` for a flow segment to use automatically, there is no way the client can control the `object_id` generated.
+Some use cases have come up with TAMS that required a need to "bring your own" `object_id`. The main use case is around TAMS Store replication. If a store is going to replicate content from another store without modification it makes sense to re-use the ids in use in the source store. Sources and Flows already allow for this but `object_id`s do not.
+
+Currently the nearest option to allow this use case involves performing operations to the backend TAMS infrastructure outside of the API as such is not a viable option.  
+
+## Considered Options
+
+- Option 1: Do nothing
+- Option 2: Change the validation on the segments endpoint
+- Option 3: Add capability to the storage endpoint to allow `object_id`s to be supplied
+
+## Decision Outcome
+
+tbc
+
+### Implementation
+
+See the API specification changes in PR [#132](https://github.com/bbc/tams/pull/132).
+
+## Pros and Cons of the Options
+
+### Option 1: Do nothing
+
+Make no changes, replication use cases should use new `object_id`s. The combination of the Flow Id and Timerange being the same as the source store will have to suffice.
+
+- Good: No changes to the API spec required
+- Good: Object Validation remains robust
+- Bad: Feels like it is not addressing the need for replication. Since different `object_id`s may imply different content.
+
+### Option 2: Change the validation on the segments endpoint
+
+Update the validation on the POST segments endpoint to not enforce `object_id`s needing to be "known". One option could be to do this by requiring the client to supply a "special" get_url so that it can be recognised as a special case and `object_id` validation is not required.
```

Comment:
The segments endpoint will already accept `object_id`s that aren't known. Specifically, objects that are known must be registered against the flow they were created against on first registration. Completely unknown IDs are permitted to support things like reference URLs that already exist in other systems. This option still won't support the allocation of storage with custom object IDs.